### PR TITLE
Fix the 4.4.x build

### PR DESCRIPTION
**Pull request by gkalpak**
This PR backports (and adapts) relevant commits from #18722, #19511 and #19603 to 4.4.x.
It also builds on top of #19586 to fix the aio tests.
Finally it adds a couple of commits on top of that to fix `NgPackagesInstaller` testing and some 4.4.x-specific issues.

### CODE REVIEW

File: aio/package.json [hunk 1]
```diff
@@ -6,33 +6,45 @@
...
-    "example-e2e": "node ./tools/examples/run-example-e2e",
+    "example-e2e": "yarn example-check-local && node ./tools/examples/run-example-e2e",
     "example-lint": "tslint -c \"content/examples/tslint.json\" \"content/examples/**/*.ts\" -e \"content/examples/styleguide/**/*.avoid.ts\"",
+    "example-use-local": "node tools/ng-packages-installer overwrite ./tools/examples/shared",
+    "example-use-npm": "node tools/ng-packages-installer restore ./tools/examples/shared",
+    "example-check-local": "node tools/ng-packages-installer check ./tools/examples/shared",
     "deploy-preview": "scripts/deploy-preview.sh",
     "deploy-production": "scripts/deploy-to-firebase.sh",
-    "check-env": "node scripts/check-environment",
+    "check-env": "yarn ~~check-env",
```
Comment (petebacondarwin) [inline aio/package.json]: does this need a bit of `yarn install`?

**Review by petebacondarwin — APPROVED** [review]

**Comment by petebacondarwin** [comment]
Needs some approval from the `build-and-ci` group.

File: aio/package.json (hunk 1 above)
Comment (gkalpak) [reply aio/package.json]: Yeah. I thought I applied #19603 on top of the other changes. Good catch :smiley:

**Comment by petebacondarwin** [comment]
Travis e2e tests for AIO are not happy :-(

**Comment by gkalpak** [comment]
It's the flakey upgrade tests 😞 They pass locally 😕 
I'll see what I can do...

**Comment by gkalpak** [comment]
I ended up disabling the failing e2e tests (after I realized they are disabled on master too). It is some typings-related issue. (Opened #19625 to track it.)

**Comment by chuckjaz** [comment]
This PR has 16 commits and many don't look related to the fixing the PR. Can the commits that are covered by other PRs be removed from this?

**Comment by gkalpak** [comment]
@chuck, not sure which commits you refer to. The only commit that is covered by another PR is the one I brought in from #19586. It will be a "no-op" if #19586 is merged first.

Other than that, this PR backports other PRs that have landed on master (which are necessary to correctly run the examples e2e tests against the local Angular build) and adds some fixes on top (some of them 4.4.x specific).

**Comment by gkalpak** [comment]
This has been merged into 4.4.x.

**Pull request closed without merging**